src/neural/cuda/layers.cc

```diff
@@ -1204,6 +1204,12 @@ void ResidualBlock<DataType>::Eval(int N, DataType* output,
   DataType* transformed_output =
       transformed_input + scratch_size / (2 * sizeof(DataType));
```

Comment:
Not sure how I feel about explicitly performing pointer math with a null variable here - even if it is always discarded